### DISCUSSION TITLE
Using Lightkurve to calculate the apparent magnitude of Algol

### DISCUSSION THREAD

**Discussion by gloglo17** [lightkurve/lightkurve#1337] [Q&A]
Hello,

I am interested in using Lightkurve to calculate the apparent magnitude of Algol for a specific date. Could someone please help me with this?

I have looked through the Lightkurve documentation but I am still unsure how to proceed. Could someone please provide me with an example code snippet or point me towards the relevant resources?

Thank you in advance for your help.

**Comment by warrickball** [comment]
I presume you've set up Python, installed Lightkurve and can import it. Here's one example of how you might go about calculating Algol's apparent magnitude. 

The first step is to search for data with something like
```
>>> import lightkurve as lk
>>> lk.search_lightcurve('algol')
No data found for target "algol".
```
Hmmm... I've found that for very bright stars, the co-ordinates to which the name resolves sometimes aren't that close to the co-ordinates in the TESS catalogue, so nothing is found. One solution is to look up the TIC ID on a service [like SIMBAD](http://simbad.u-strasbg.fr/simbad/sim-basic?Ident=algol), where you can see it's TIC 346783960. Alternatively, you can increase the search radius a bit. e.g.
```
>>> lk.search_lightcurve('algol', radius=3)
SearchResult containing 4 data products.

 #     mission     year   author  exptime target_name distance
                                     s                 arcsec 
--- -------------- ---- --------- ------- ----------- --------
  0 TESS Sector 18 2019      SPOC     120   346783960      1.9
  1 TESS Sector 58 2022      SPOC     120   346783960      1.9
  2 TESS Sector 18 2019 TESS-SPOC    1800   346783960      2.9
  3 TESS Sector 18 2019       QLP    1800   346783960      2.9
```
where we can also see that the star is TIC 346783960 (but this is always worth cross-checking). You can now try downloading and plotting some of this data with e.g.
```
>>> lc = lk.search_lightcurve('algol', radius=3)[0].download()
>>> lc.plot()
```
Depending how your Python environment is set up, you might see the plot or explicitly call `matplotlib.pyplot.show()`. I get this:
![image](https://github.com/lightkurve/lightkurve/assets/20858744/76ad6b66-cfe8-4fb6-8bc3-77fc924f25c9)
The thing now is to convert the flux (in electrons per second) to a magnitude. We estimate this by converting the flux variations into magnitude variations:
```
>>> import numpy as np
>>> delta_mag = -2.5*np.log10(lc.flux.value/np.nanmedian(lc.flux.value))
```
and adding that to the magnitude in the metadata (about 3.37):
```
>>> mag = lc.meta['TESSMAG'] + delta_mag
```
You'll only have the magnitude at the times in the TESS data (Sectors 18 and 58, according to our search). Also, I expect a bright target like Algol is heavily saturated and the brightness variations might not be reliable. But how precise you need it to be depends on what you're trying to do. I routinely use saturated stars as examples of light curve shapes. I see SIMBAD gives Algol R-, I- and J-band magnitudes of about 2.1, so the TESS magnitude seems a bit high.
